### PR TITLE
Disable password field on block page if no password is set

### PR DESCRIPTION
**Pull request by Mcat12**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

10

---

For when you want to whitelist a domain and no password is set.
Requires pi-hole/AdminLTE#597

screenshot from 2017-10-07 16-50-02

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: advanced/index.php [hunk 1]
```diff
@@ -209,11 +209,17 @@ function queryAds($serverName) {
     window.onload = function () {
       <?php
       // Remove href fallback from "Back to safety" button
-      if ($featuredTotal > 0) echo '$("#bpBack").removeAttr("href");';
-      // Enable whitelisting if $svPasswd is present & JS is available
-      if (!empty($svPasswd) && $featuredTotal > 0) {
-          echo '$("#bpWLPassword, #bpWhitelist").prop("disabled", false);';
+      if ($featuredTotal > 0) {
```
Comment (WaLLy3K) [inline advanced/index.php]: This was a single line statement, why does this now cover the other statement?

**Comment by WaLLy3K** [comment]
From a minimalist standpoint, having the password field stay there when there's no password seems tacky and unnecessary. From a "I don't understand why users on my network are able to whitelist stuff" use-case, it kinda makes sense, but *probably* should be a notification for elsewhere.

I'm of two minds on this matter - what are your thoughts?

File: advanced/index.php (hunk 1 above)
Comment (Mcat12) [reply advanced/index.php]: Now the whitelist button is always enabled (as long as `$featuredTotal > 0`), but the password input is only enabled if there's a password.

**Comment by Mcat12** [comment]
Do you mean you would want to remove the password input when there's no password, and put some notice on the web interface that there is no security over the whitelist? Keeping the password input reminds them that you can limit access to the whitelist.

**Comment by WaLLy3K** [comment]
> Do you mean you would want to remove the password input when there's no password, and put some notice on the web interface that there is no security over the whitelist?

Yeah, I think that's the better option as long as the notice can be dismissed - some people would be happy to accept the "risks", especially if they're the sole user of Pi-hole.

> Keeping the password input reminds them that you can limit access to the whitelist.

It reminds the *admin*, sure, but it's irrelevant (and due to the wording, potentially confusing) to a user who is accessing the block page.

**Comment by technicalpyro** [comment]
you gentlemen are the experts but would it be possible to have a whitelist button that then either causes a  dialog for password entry of if no password is set just whitelists ?

**Comment by Mcat12** [comment]
Do you want that just as a UI change? What benefits would that bring?

**Comment by technicalpyro** [comment]
> Do you want that just as a UI change? What benefits would that bring?

in my mind it would solve some of the present problem either way you have a button to whitelist if the admin password is set it then asks for it if not it just does the whitelist procedure.  unless i was misunderstanding the discussion above and way off in left field

**Comment by Mcat12** [comment]
It might make sure the user doesn't get confused, sure.

**Comment by WaLLy3K** [comment]
Currently getting the following error: 

> No password set - whitelisting without password not supported

I believe this message is pulled from `/admin/scripts/pi-hole/php/add.php`, but I thought we already merged an update that fixed this?

**Comment by Mcat12** [comment]
Are you on the correct web branch? This requires devel.

**Comment by Mcat12** [comment]
@WaLLy3K What do you think of this PR now?

**Comment by WaLLy3K** [comment]
Functionality wise, it looks good. My only issue is that removing the password field hides the "[Javascript disabled](https://github.com/pi-hole/pi-hole/blob/009fa2f1b7f44ce983eece60a19ec1c109dcdcfe/advanced/index.php#L89)" text, which is used to tell a user that they need to enable JS before whitelisting. As it stands, the "Whitelist" button is just grey with no prompt as to why - this could create user confusion.

**Comment by Mcat12** [comment]
Do you want me to add the password box back so it's always there, but disabled unless there's a password?

**Comment by WaLLy3K** [comment]
That might be the easiest solution for now! 😄

**Comment by WaLLy3K** [comment]
With JS disabled, I'm only getting the "No Admin Password Set" text, and not the Enable Javascript text, which is a bit odd. Has there been any changes to this functionality that could account for why this isn't working correctly? Also, if it's possible to remove the password field when JS *is* enabled, I think that'd be great. :smile: 

untitled

**Comment by Mcat12** [comment]
The reason it's showing no password set is because of that line you referenced which decides which message to show by default based on the presence of a password.

**Pull request merged**